security-command-center/snippets/src/test/java/management/api/EventThreatDetectionCustomModuleTest.java

```
@@ -0,0 +1,166 @@
+  private static final int MAX_ATTEMPT_COUNT = 3;
+  private static final int INITIAL_BACKOFF_MILLIS = 120000; // 2 minutes
+
+  @Rule
+  public final MultipleAttemptsRule multipleAttemptsRule =
+      new MultipleAttemptsRule(MAX_ATTEMPT_COUNT, INITIAL_BACKOFF_MILLIS);
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @BeforeClass
+  public static void setUp() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("SCC_PROJECT_ID");
+  }
+
+  @AfterClass
+  public static void cleanUp() throws IOException {
+    // Perform cleanup after running tests
+    cleanupExistingCustomModules();
+  }
+
+  // cleanupExistingCustomModules clean up all the existing custom module
+  private static void cleanupExistingCustomModules() throws IOException {
+    try (SecurityCenterManagementClient client = SecurityCenterManagementClient.create()) {
+      ListEventThreatDetectionCustomModulesRequest request =
+          ListEventThreatDetectionCustomModulesRequest.newBuilder()
+              .setParent(String.format("projects/%s/locations/global", PROJECT_ID))
+              .build();
+      ListEventThreatDetectionCustomModulesPagedResponse response =
+          client.listEventThreatDetectionCustomModules(request);
+      // Iterate over the response and delete custom module one by one which start with
+      // java_sample_custom_module
+      for (EventThreatDetectionCustomModule module : response.iterateAll()) {
+        if (module.getDisplayName().startsWith("java_sample_custom_module")) {
+          String customModuleId = extractCustomModuleId(module.getName());
+          deleteCustomModule(PROJECT_ID, customModuleId);
+        }
+      }
+    }
```

nit: consider silencing exception. it is better to leave leftovers with and log the problem than fail a lot of tests.